src/hardware/serialport/directserial.cpp

```diff
@@ -262,7 +262,8 @@ void CDirectSerial::updatePortConfig(uint16_t divider, uint8_t lcr)
 	uint8_t bytelength = (lcr & 0x3) + 5;
 
 	// baudrate
-	const uint32_t baudrate = divider ? 115200 / divider : 115200;
+	const uint32_t baudrate = divider ? SERIAL_MAX_BAUDRATE / divider
+	                                  : SERIAL_MAX_BAUDRATE;
```

Comment:
This code here is a dupe of the `GetPortBaudRate()` function.

Suggest de-duping them using a common static function with this logic, and use it in both/all places.